monitoring/db_collector.go

```diff
@@ -0,0 +1,81 @@
+package monitoring
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	dbSizeMetric = "total_db_size"
+)
+
+// dbCollector is a Prometheus collector that exports metrics related to the
+// daemon's database.
+type dbCollector struct {
```

Comment:
Do we really need two collector structs and files? IMO we could merge the asset proof and db collector into one (`dbCollector`) as both read things from the DB...